packages/charts/src/vaadin-chart.js

```diff
@@ -918,24 +915,33 @@ class Chart extends ElementMixin(ThemableMixin(PolymerElement)) {
     };
   }
 
-  /** @private */
-  __reflow() {
+  /**
+   * Implements resize callback from `ResizeMixin`
+   * to reflow when the chart element is resized.
+   * @protected
+   * @override
+   */
+  _onResize(contentRect) {
     if (!this.configuration) {
       return;
     }
-    this.configuration.reflow();
-  }
 
-  /** @private */
-  __mutationCallback() {
-    const { height: componentHeight } = this.getBoundingClientRect();
-    const { chartHeight } = this.configuration;
+    const { height, width } = contentRect;
+    const { chartHeight, chartWidth } = this.configuration;
 
-    if (componentHeight !== chartHeight) {
+    if (height !== chartHeight || width !== chartWidth) {
       this.__reflow();
     }
   }
 
+  /** @private */
+  __reflow() {
+    if (!this.configuration) {
+      return;
+    }
```

Comment:
nit: this already seems to be checked in both places where `__reflow` is called.

Reply:
I would prefer to keep it here just in case someone tries to call the method manually. Also, if we used TS for type checking, this guard would be needed.